core/src/main/scala/cr/pulsar/Producer.scala

```diff
@@ -34,6 +34,16 @@ trait Producer[F[_], E] {
     */
   def send(msg: E): F[MessageId]
 
+  /**
+    * Sends a delayed message asynchronously. Works only with shared subscriptions.
+    */
+  def sendDelayed(msg: E, delay: FiniteDuration): F[MessageId]
+
+  /**
+    * Sends a delayed message asynchronously but it discards its output. Works only with shared subscriptions.
+    */
+  def sendDelayed_(msg: E, delay: FiniteDuration): F[Unit]
+
```

Comment:
It can be useful to add `cancel` method which simply `ack` the provided `MessageId`

Reply:
I'm not sure that's possible (at least don't see it in Java API). Anyway, if we will need it - will look for a way to do it, but for now it's not required